woocommerce/woocommerce-android
# Jetpack CP: Handle various plugin installation failure cases.

### Description


This PR adds error screen on two separate cases:
1) Installation failure (after automated retries),
2) Activation failure (after automated retries)

During installation process, if the app encounters installation or activation error, the first thing it does is to automatically retry a couple of times (this is already added/tested in [a previous PR](https://github.com/woocommerce/woocommerce-android/pull/5360)). If the error persists after a retry, then it will show an error message, plus a couple of buttons to let merchants to:
- install / activate directly in wp-admin
- contact support

Additionally, this also adds tracks for the install button and contact support button.

### What is not included in this PR
- The next step after plugin activation is the "connecting" step on the UI, where behind the scene what is done is to check the completion of full sync status after Jetpack is activated. This will be added in a future PR.
- Once merchant installs and activates Jetpack on wp-admin and then returns to the app, the app has to intercept that and continues with the "connecting" step. This will also be added in a future PR (as it relies on the full sync work above).

### Images


| Installation failure | Activation failure |
|-|-|
| <img src="https://user-images.githubusercontent.com/266376/147223073-0a8e11a7-06ea-4af2-9ad2-44fd18101ab1.png" /> | <img src="https://user-images.githubusercontent.com/266376/147223400-d92a2ffc-582e-4c00-bbd4-aea974be0b8f.png" /> |


### Testing instructions


Make sure to have a JCP site ready: A site with WooCommerce and WooCommerce Payments that's already connected to a WordPress.com account, and no Jetpack plugin.

**A. Testing error message for installation failure:**
1. Go to `JetpackInstallViewModel` class,
2. Replace all content inside `init` with `viewState = viewState.copy(installStatus = Failed(INSTALLATION, "Error"))`
3. Run app, open the test site.
4. Tap the Jetpack banner and go through installation process.
5. The install screen will immediately show the error screen, similar to the first screenshot above.
6. Tap the Install Jetpack in WP-Admin button, and make sure it opens to "wp-admin/plugin-install.php?tab=plugin-information&plugin=jetpack" once you log in to wp-admin
7. Close the Chrome tab, return to the app,
8. Tap the Contact Support button, and make sure it opens to the support screen

**B. Testing error message for activation failure:**
1. Go to `JetpackInstallViewModel` class,
2. Replace all content inside `init` with `viewState = viewState.copy(installStatus = Failed(ACTIVATION, "Error"))`
3. Run app, open the test site.
4. Tap the Jetpack banner and go through installation process.
5. The install screen will immediately show the error screen, similar to the second screenshot above.
6. Tap the Activate Jetpack in WP-Admin button, and make sure it opens to "wp-admin/plugin.php" once you log in to wp-admin
7. Close the Chrome tab, return to the app,
8. Tap the Contact Support button, and make sure it opens to the support screen
